crates/engine_bibtex/bibtex/bibtex.c

```diff
@@ -7,1929 +7,166 @@
+        break;
+    }
+    return res.ok;
 }
 
-static bool scan3(ASCII_code char1, ASCII_code char2, ASCII_code char3)
-{
-    buf_ptr1 = buf_ptr2;
-    while ((buf_ptr2 < last) && (buffer[buf_ptr2] != char1) && (buffer[buf_ptr2] != char2) && (buffer[buf_ptr2] != char3))
-        buf_ptr2 = buf_ptr2 + 1;
-    return buf_ptr2 < last;
-}
+/*:473*//*12: *//*3: */
 
-static bool scan_alpha(void)
-{
-    buf_ptr1 = buf_ptr2;
-    while ((buf_ptr2 < last) && (lex_class[buffer[buf_ptr2]] == 2 /*alpha */ ))
-        buf_ptr2 = buf_ptr2 + 1;
-    return (buf_ptr2 - buf_ptr1) != 0;
-}
+#define FMT_BUF_SIZE 1024
+static char fmt_buf[FMT_BUF_SIZE] = "";
 
-static void scan_identifier(ASCII_code char1, ASCII_code char2, ASCII_code char3)
+PRINTF_FUNC(1,2) static void
+printf_log(const char *fmt, ...)
 {
-    buf_ptr1 = buf_ptr2;
-    if (lex_class[buffer[buf_ptr2]] != 3 /*numeric */ )
-        while ((buf_ptr2 < last) && (id_class[buffer[buf_ptr2]] == 1 /*legal_id_char */ ))
-            buf_ptr2 = buf_ptr2 + 1;
-    if ((buf_ptr2 - buf_ptr1) == 0)
-        scan_result = 0 /*id_null */ ;
-    else if ((lex_class[buffer[buf_ptr2]] == 1 /*white_space */ ) || (buf_ptr2 == last))
-        scan_result = 3 /*white_adjacent */ ;
-    else if ((buffer[buf_ptr2] == char1) || (buffer[buf_ptr2] == char2) || (buffer[buf_ptr2] == char3))
-        scan_result = 1 /*specified_char_adjacent */ ;
-    else
-        scan_result = 2 /*other_char_adjacent */ ;
-}
+    va_list ap;
 
-static bool scan_nonneg_integer(void)
-{
-    buf_ptr1 = buf_ptr2;
-    token_value = 0;
-    while ((buf_ptr2 < last) && (lex_class[buffer[buf_ptr2]] == 3 /*numeric */ )) {
+    va_start (ap, fmt);
+    vsnprintf (fmt_buf, FMT_BUF_SIZE, fmt, ap);
+    va_end (ap);
 
-        token_value = token_value * 10 + (buffer[buf_ptr2] - 48);
-        buf_ptr2 = buf_ptr2 + 1;
-    }
-    return (buf_ptr2 - buf_ptr1) != 0;
+    puts_log(fmt_buf);
 }
 
-static bool scan_integer(void)
-{
-    unsigned char sign_length;
-    buf_ptr1 = buf_ptr2;
-    if (buffer[buf_ptr2] == 45 /*minus_sign */ ) {
-        sign_length = 1;
-        buf_ptr2 = buf_ptr2 + 1;
-    } else
-        sign_length = 0;
-    token_value = 0;
-    while ((buf_ptr2 < last) && (lex_class[buffer[buf_ptr2]] == 3 /*numeric */ )) {
-
-        token_value = token_value * 10 + (buffer[buf_ptr2] - 48);
-        buf_ptr2 = buf_ptr2 + 1;
-    }
-    if (sign_length == 1)
-        token_value = -(int32_t) token_value;
+static void log_fprintf(const char* fmt, ...) {
+    va_list ap;
+    va_start(ap, fmt);
+    vsnprintf(fmt_buf, FMT_BUF_SIZE, fmt, ap);
+    va_end(ap);
 
-    return (buf_ptr2 - buf_ptr1) != sign_length;
+    bib_log_prints(fmt_buf);
 }
 
-static bool scan_white_space(void)
-{
-    while ((buf_ptr2 < last) && (lex_class[buffer[buf_ptr2]] == 1 /*white_space */ ))
-        buf_ptr2 = buf_ptr2 + 1;
-    return buf_ptr2 < last;
-}
+/*:159*//*160: */
 
-static bool eat_bst_white_space(void)
+static bool eat_bst_white_space(GlblCtx* ctx)
```

Comment:
This is very much not a big deal, but I don't love the mega-abbreviated name `GlblCtx` ... maybe just `Bibtex`, or `Engine`, or something like that? Or just `GlobalCtx` would be a bit nicer to my tastes.